src/server/standalone.js

```diff
@@ -386,6 +386,51 @@ function StandAloneServer(gmeConfig) {
         }
     }
 
+    function getGoodExtraAssetRouteFor(component, basePaths) {
+        // Check for good extra asset
+        return function(req, res) {
+            res.sendFile(Path.join(__baseDir, req.path), function (err) {
+                if (err && err.code !== 'ECONNRESET') {
+                    //this means that it is probably plugin/pluginName or plugin/pluginName/relativePath format
+                    // so we try to look for those in our config
+                    //first we check if we have the plugin registered in our config
+                    var urlArray = req.url.split('/'),
+                        pluginName = urlArray[2] || null,
+                        basePath,
+                        baseAndPathExist,
+                        relPath;
+
+                    relPath = getRelPathFromUrlArray(urlArray);
+                    basePath = getBasePathByName(pluginName, basePaths);
+                    baseAndPathExist = typeof basePath === 'string' && typeof relPath === 'string';
+                    if (baseAndPathExist &&
+                        webgmeUtils.isGoodExtraAsset(pluginName, Path.join(basePath, pluginName))) {
+                            expressFileSending(res, Path.resolve(Path.join(basePath, relPath)));
+                    } else {
+                        res.sendStatus(404);
+                    }
+                }
+            });
+        };
+    }
+
+    function getRelPathFromUrlArray(urlArray) {
+        urlArray.shift();
+        urlArray.shift();
+        urlArray.shift();
+        var relPath = urlArray.join('/');
+        if (!Path.extname(relPath)) {  // js file by default
+            relPath += '.js';
+        }
+        return relPath;
+    }
+
+    /**
+     * Unlike `getGoodExtraAssetRouteFor`, `getRouteFor` does not assume that the 
+     * resource hosts a main file which has the same structure as the parent directory.
+     * That is, there are examples of panels (such as SplitPanel) in which the
+     * main file does not adhere to the format "NAME/NAME+'Panel'"
```

Comment:
If there is no common rule for that format, how can it know which file to require?